### PR TITLE
Seed tests

### LINKED ISSUE #865: E2E seed tests
### Tasks

According to doc/usecases/*/entry_list_seeds.md

- [x] Test seed creation
- [x] Test seed editing
- [x] Test seed ~~deletion~~ archiving
- [x] Test seed search
- [x] Test seed undo delete
- [ ] Test error case
   - [ ] Invalid expiration date
   - [ ] Same name

### PR DESCRIPTION
**Pull request by 4ydan**
Closes #865
I think it was too early to write these tests as many issues exist and the usecase is not `done` yet.
Therefore I will mark this as draft and wait before finishing the tests.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by markus2330** [comment]
> I think it was too early to write these tests as many issues exist and the usecase is not done yet.
> Therefore I will mark this as draft and wait before finishing the tests.

#836 is now merged, so it is ready to be tested now. Pls rebase :rocket:

**Comment by 4ydan** [comment]
- Seed delete doesnt exist.
- Seed edit doesnt exist.
- Validation is unfinished.

It is not possible to continue efficiently for now.

**Comment by markus2330** [comment]
>  Seed delete doesnt exist.
>  Seed edit doesnt exist.

Do you mean the locators? The UI elements are there and work for me:

image

The only weirdness is that for "delete" it seems to somewhat refresh the whole UI and you get greeted again. If this is a problem for e2e tests, please create a release-critical issue. Otherwise I think it is only minor annoyance. I added it to #872

>  Validation is unfinished.

Yes, this is tracked in #868

**Comment by 4ydan** [comment]
I am sorry, somehow I didnt see the latest version. Now it works.

File: e2e/pages/seeds/create.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self._notes = page.get_by_test_id("notest")
+
+    def create_a_seed(self):
+        """
+        Helper function to create a seed
+        Fills out fields and clicks create at the end
+        which navigate to the `SeedManagementPage`
+        """
+        # Plant Name TODO use better locator
+        self._page.get_by_role("main").locator("svg").first.click().click()
```
Comment (4ydan) [inline e2e/pages/seeds/create.py:37]: I really dont know how to add this locator in `frontend/`

File: e2e/pages/seeds/create.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        self._page.get_by_text("Abelia triflora (indian abelia)", exact=True).click()
+
+        # Amount TODO use better locator
+        self._amount.click()
```
Comment (4ydan) [inline e2e/pages/seeds/create.py:41]: And this one too.

File: frontend/src/features/seeds/components/CreateSeedForm.tsx [hunk 3]
```diff
@@ -172,6 +172,7 @@ const CreateSeedForm = ({
                 setPlantOption(option as SelectOption);
               }}
               onChange={onChange}
+              data-testid="plant_name"
```
[the remark this answers is not in the document]
Comment (4ydan) [reply frontend/src/features/seeds/components/CreateSeedForm.tsx:175]: image

**Comment by markus2330** [comment]
@badnames can you please help out with these 4 locators?

File: frontend/src/features/seeds/components/CreateSeedForm.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (markus2330) [reply frontend/src/features/seeds/components/CreateSeedForm.tsx:175]: @badnames please help :cry:

**Comment by 4ydan** [comment]
Similar https://github.com/testing-library/react-testing-library/issues/377

**Comment by 4ydan** [comment]
Its the best if I finish this after #872 is done and maybe the search is also enabled. I think there is still some open questions that will change how the tests will look, which I have asked in https://github.com/ElektraInitiative/PermaplanT/issues/872#issuecomment-1723288292

**Comment by markus2330** [comment]
I think it is best to get this merged asap, with what currently works, so that it is also the responsibility of the people who change something to look for regressions.

When the search gets added, we add more tests.

**Comment by 4ydan** [comment]
rebased on #973, hence the messy commits and changed files

File: e2e/features/seeds.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
+Feature: Seed Creation
+    As a user I want to be able to create seeds
+
+    Background:
+        Given I am on the seed management page
+
+    Scenario: Successful seed creation
+        When I create a new seed
+        Then I can successfully create a new seed without an error message
```
Comment (markus2330) [inline e2e/features/seeds.feature:9]:
```suggestion
        Then the seed is shown on the overview page
```

File: e2e/features/seeds.feature [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    Scenario: Successful seed editing
+        When I create another new seed
+        And I edit this seed
+        Then the edited seed is saved and displayed correctly
```
Comment (markus2330) [inline e2e/features/seeds.feature:14]:
```suggestion
        Then the edited seed is saved and shown on the overview page
```

File: e2e/features/seeds.feature [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    Scenario: Successful seed search
+        When I search for a seed
+        Then I can see the seed in the table
```
Comment (markus2330) [inline e2e/features/seeds.feature:18]:
```suggestion
        Then I can see the seed in the overview page
```

File: e2e/features/seeds.feature [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+    Scenario: Searching seed that does not exists
+        When I search for a seed that does not exist
+        Then the search result is empty
+
+    Scenario: Archving a seed
+        When I try to archive a seed
+        Then the seed dissapears
```
Comment (markus2330) [inline e2e/features/seeds.feature:26]:
```suggestion
        Then the seed disappears
```

File: e2e/pages/seeds/create.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        Fills out fields and clicks create at the end
+        which navigate to the `SeedManagementPage`
+        """
+        self._plant_name.click()
+        self._page.get_by_text(plant_name, exact=True).click()
+
+        self._amount.click()
+        self._page.get_by_text(amount, exact=True).click()
+
+        # TODO, needs better locator, not working
```
Comment (markus2330) [inline e2e/pages/seeds/create.py:40]: better locator for what?

**Review by markus2330 — APPROVED** [review]
Great job. A test with >30 seeds would be important (as this is too time intensive to test manually). Can you create a left-over or add it?

**Comment by 4ydan** [comment]
I removed two timeline tests because they were broken in some other PR.

File: e2e/pages/seeds/create.py (hunk 8 above)
Comment (4ydan) [reply e2e/pages/seeds/create.py]: Done, works now, I cant remember.

**Comment by markus2330** [comment]
Is this to be merged before #1063?

**Comment by 4ydan** [comment]
I dont know. I dont think it interferes with it.

**Pull request merged**